once_test.py

```diff
@@ -77,13 +156,30 @@ def test_different_args_same_result(self):
         self.assertEqual(counter.value, 1)
 
     def test_iterator(self):
+        counter = Counter()
+
         @once.once
         def yielding_iterator():
-            for i in range(3):
-                yield i
+            nonlocal counter
+            for _ in range(3):
+                yield counter.get_incremented()
+
+        self.assertEqual(list(yielding_iterator()), [1, 2, 3])
+        self.assertEqual(list(yielding_iterator()), [1, 2, 3])
+
+    def test_iterator_parallel_execution(self):
+        counter = Counter()
+
+        @once.once
+        def yielding_iterator():
+            nonlocal counter
+            for _ in range(3):
+                yield counter.get_incremented()
 
-        self.assertEqual(list(yielding_iterator()), [0, 1, 2])
-        self.assertEqual(list(yielding_iterator()), [0, 1, 2])
+        with concurrent.futures.ThreadPoolExecutor(max_workers=16) as executor:
```

Comment:
nit: might want to abstract max_workers as an env variable or flag, or at least a constant that's reused somewhere, so we can easily tweak it.

Reply:
will do in upcoming cleanup PR.